### PR TITLE
Add composite foreign key constraints generation when baking a snapshot

### PR DESCRIPTION
**Pull request by HavokInspiration**
Since composite foreign keys constraints support was added in CakePHP schema reflection, baked migrations should support them as well.

### CODE REVIEW

**Comment by HavokInspiration** [comment]
I think the failing build is due to the fact that it tests the 3.0 initial release which did not have the composite FK constraints support.
The problem is located where the ``quoteIdentifier()`` method is called and an array of column is passed to it where creating the ``orders`` fixtures.

I'm thinking about creating a separate test for this feature and skip it based on CakePHP core version. Any ideas / other suggestions ?

**Comment by HavokInspiration** [comment]
I added a new test that is skipped if CakePHP Core version is lower than 3.0.8

**Comment by lorenzo** [comment]
@HavokInspiration Good idea!

File: tests/TestCase/Shell/Task/MigrationSnapshotTaskTest.php [hunk 1]
```diff
@@ -77,4 +90,23 @@ public function testNotEmptySnapshot()
         $result = $this->Task->bake('NotEmptySnapshot');
         $this->assertSameAsFile(__FUNCTION__ . '.php', $result);
     }
+
+    public function testCompositeConstraintsSnapshot()
+    {
+        $cakeVersion = intval(str_replace('.', '', Configure::version()));
```
Comment (lorenzo) [inline tests/TestCase/Shell/Task/MigrationSnapshotTaskTest.php]: there is a php function call `version_compare()` maybe use that instead?

**Comment by HavokInspiration** [comment]
@lorenzo Done.

**Comment by lorenzo** [comment]
great job @HavokInspiration !

**Pull request merged**